fill in installing software and browsing the web (#37456)
mdn/content: files/en-us/learn_web_development/getting_started/environment_setup/browsing_the_web/index.md

{{PreviousMenuNext("Learn_web_development/Getting_started/Environment_setup/Installing_software", "Learn_web_development/Getting_started/Environment_setup/Code_editors", "Learn_web_development/Getting_started/Environment_setup")}}

> By this point in the module, you should have at some modern web browsers installed on your computer and other available devices. This article goes a little deeper into using browsers, looking at how a web browser works, the difference between some of the common items you'll interact with, and how to search for information.

Reviewer: ```suggestion
By this point in the module, you should have a modern web browser installed on your computer or other available devices. This article goes deeper into using browsers, looking at how a web browser works, the difference between some of the everyday things you'll interact with, and how to search for information.
```

Author: I used this, except that I replaced "a modern web browser" with "multiple modern web browsers" — in the installing software article, I make it clear that I think readers should have multiple browsers installed, to try to promote cross-browser testing.

Reviewer: OK, sounds good

> [!NOTE]
> If you don't have any browsers installed beyond the default ones that came with your devices, get some more installed. See [Modern web browsers](/en-US/docs/Learn_web_development/Getting_started/Environment_setup/Installing_software#modern_web_browsers).

> As with any area of knowledge, the web comes with a lot of jargon. Don't worry. We won't overwhelm you with all of it (we have a [glossary](/en-US/docs/Glossary) if you're curious). However, there are some basic terms you need to understand at the outset since you'll hear these expressions all the time as you read on. We introduce some important terms below.

Reviewer: ```suggestion
As with any area of knowledge, the web comes with a lot of jargon and technical terminology. Don't worry. We won't overwhelm you with all of it (you can check the [glossary](/en-US/docs/Glossary) if you're curious). However, there are some basic terms you need to understand from the beginning since you'll hear these expressions all the time as you read on. We introduce some important terms below.
```

Author: I mostly liked this, but tweaked it a little:

> As with any area of knowledge, the web comes with a lot of jargon and technical terminology. Don't worry: We won't overwhelm you with all of it upfront (you can check the [glossary](/en-US/docs/Glossary) if you're curious). However, there are some basic terms you need to understand from the beginning since you'll hear these expressions all the time. We introduce some important terms below.

Reviewer: great, ty

      <th scope="row">Learning outcomes:</th>
      <td>
        <ul>
          <li>The difference between a web browser, a website, and a search engine.</li>
          <li>How a web browser works at a basic level.</li>
          <li>Searching for information.</li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>

## The difference between web page, website, web server, and search engine

We will start by describing various web-related concepts: web pages, websites, web servers, and search engines. These terms are often confused by newcomers to the web or are incorrectly used. Let's make sure you know what they each mean! Let's start with some definitions:

- **Web page**
  - : A document that can be displayed in a web {{Glossary("browser")}}. These are also often called just "pages." Such documents are written in the {{Glossary("HTML")}} language (which we look at in more detail later on).
- **Website**

>   - : A collection of web pages grouped together into a single resourse, with links connecting them together. Often called a "site."

Reviewer: Are we doing `"site."` instead of `"site".`? 😄

Author: Oh lord no! This was here in the original text, you gotta believe me!

I've fixed this, and the earlier instance of `"pages."` ;-)

Reviewer: oh thank goodness

### Search engine

Search engines are a common source of confusion on the web. A search engine is a special kind of website that helps users find web pages from _other_ websites.

There are plenty out there: [Google](https://www.google.com/), [Bing](https://www.bing.com/), [Yandex](https://yandex.com/), [DuckDuckGo](https://duckduckgo.com/), and many more. Some are generic, some are specialized around certain topics.

Many beginners on the web confuse search engines and browsers. Let's make it clear: A _browser_ is a piece of software that retrieves and displays web pages; a _search engine_ is a website that helps people find web pages from other websites. The confusion arises because, the first time someone launches a browser, the browser often displays a search engine's homepage or a search box allowing them to search for a term using that search engine. Most browsers also allow their users to use a search engine by typing search terms directly into the browser address bar.

> This all makes sense because the first thing people tend to want to do with a browser is find a web page to display. Don't confuse the infrastructure (the browser) with the service (the search engine).

Reviewer: I'm not sure about 'infrastructure' usage here:

```suggestion
This all makes sense because the first thing people tend to want to do with a browser is find a web page to display. Don't confuse the software (the browser) with the service (the search engine).
```

You could also use "car and road" metaphor, but the road is more than a search engine :)

Here is an instance of Firefox showing a Google search box as its default startup page:

![Example of Firefox nightly displaying a custom Google page as default](search-engine.jpg)

Try it out now — do a search in a search engine by:

- Going to a search engine homepage and entering a search term.
- Entering a search term into the browser address bar.

> - Entering a search term somewhere else in the browser.

Reviewer: ```
- Entering a search term somewhere else in the browser.
```

Where else?

Reviewer: >other dedicated parts of a browser UI for doing searches are few and far between, at least by default

Yeah I was struggling to think of something, sounds good tnx

## Searching for information

As a web developer, you will spend a lot of time searching for information, from syntax you can't remember to solutions to specific problems. It is therefore a good idea to learn how to effecively search the web.

If you are looking for general information about a specific web technology feature, you should type the name of the feature into the MDN search box. For example, try typing `box model`, `fetch()` or `video element` into the the search box and see what comes up. If you don't find the information you need, try expanding your search — try your search term in a search engine.

If you are looking for a solution to a specific problem, such as `how to print out the fibonacci sequence with JavaScript` or `how to calculate whether a number is a prime number with JavaScript`, it is a good idea to search on a website such as [StackOverflow](https://stackoverflow.com), which is a community dedicated to answering programming problems. Again, try using a general search engine if a specific site doesn't give you a helpful answer.

> 
> Try some searches, as indicated above.

Reviewer: Maybe remove?

```suggestion
```

Author: Maybe we can get the dev team to implement a custom style for try it out sections?

Reviewer: oh, I see. Maybe we're looking for a callout?

```md
> [!CALLOUT]
>
> **A callout.**
>
> It can have multiple paragraphs.
```

like

```md
> [!CALLOUT]
>
> **Try it out**
```

When learning to code, AI chat prompts can be useful in a variety of ways:

- Doing conventional searches, like the examples above.
- Figuring out bugs in a block of code. If you are getting frustrated because your code is not working, you can paste your code into an AI chat prompt, preceeded by a question such as `Where is the mistake in this code?`
- Generating an optimized version of a specific block of code. This can be useful when you've written a block of code that works, but you want to find out how it could be done more efficiently, or in a more robust way that solves more use cases.
- Providing advice on how to do something. For example, if you don't just want to know where the bug is in a block of code, but instead you want advice on what strategy to use to debug it.

> Try using a couple of AI tools to do some searches.

Reviewer: Maybe it's style/subjective, but I would remove this last sentence. Keep if you like it.

Author: See previous comment. I've again applied the same styling to this bit, so we can see what a single-sentence "Try it out" block might look like.

### A cautionary tale

In truth, AI can do so much that you may start to wonder why you need to learn to code.

But wait! The following is important: **You still need to understand what you are trying to do at a high level, what the code is doing, and where each piece of code needs to be used**. If you don't, you won't be very useful when trying to figure out what to do to solve real-world problems. This means that you still need to learn to code. AI can be a really useful tool to help you find answers more quickly, but if you just type every question you are asked into an AI prompt, you won't understand how anything works.

> In addition, AI tools present their answers in a confident, authoritative voice, but they can often be misleading or just plain wrong. Some of the errors they make can be very subtle. They don't have any innate intelligence of their own — they are basically advanced pattern matching tools. AI tools compile their answers from other sources out there, so will hoover up wrong information as well as correct information. Even two correct sources can be combined to create an answer that is incorrect.

Reviewer: Newer information may not also be available, or answers may be skewed to older and more prevalent documentation, so "how to do X in JS" might give you outdated guidance.

Author: I really like this point, so I've turned it, and the above paragraph, into two separate bullets.

> As a result, you need to be careful to check the answers they give you, and not just trust everything without question.

Reviewer: Maybe not just in this section, but I think it's good to mention:

* knowing how to look for documentation
* knowing how to read documentation (a tool or language reference, specs, MDN, GitHub READMEs, whatever)
* debugging -> knowing what output is important and how to find information about the problem
* communities where you can ask for help

as alternatives.

Reviewer: You already mention search tips below, so would link to that / mention that comes next

Author: There is an entire article later on covering [research and learning](https://developer.mozilla.org/en-US/docs/Learn_web_development/Getting_started/Soft_skills/Research_and_learning), which is where I think this stuff should go. That one also needs writing ;-)

I've noted down your points for when I do start to tackle that one. I'll also make sure to link to it from here.

**When you are learning, spend time trying to solve the problem yourself before searching for an answer, whether you are using AI or a conventional search engine. It will make you a better developer.**

### Search tips

- You should include the language you are using in the search term, as shown in the examples above. If you just typed in `how to print out the fibonacci sequence`, you would likely end up with several solutions in Python, C++, Java, Ruby, or other languages — not quite as helpful when you are trying to learn JavaScript!
- When you find a useful answer, bookmark or make a copy of it somewhere so you can find it again later. You'll be amazed how many times you run into the same problem.
- If your code is returning a specific error message, try entering the error into a search engine or AI prompt. Other people will probably have already tackled the same error in the past and recorded solutions publicly somewhere.
- If possible, stick with recommended sites like MDN and [StackOverflow](https://stackoverflow.com).

> - There are many advanced search techniques you can use in search engines that will give you better results than just typing a plain search term. Typing in a plain search term such as `ant fish cheese` will return results that contain any combination of those words. However:

Reviewer: Small one to mention it may differ per search service:

```suggestion
- There are many advanced search techniques you can use in search engines that will give you better results than just typing a plain search term. Typing in a plain search term such as `ant fish cheese` will return results that contain any combination of those words. However search engines usually support variations of the following formats:
```

Author: I like it. I've used it, but with a slight tweak:

> However, most search engines support variations of the following formats:

  - Typing in `"ant fish cheese"` (with the quotes) will only return results that contain that exact phrase.
  - `"ant cheese" -fish` will return results that contain `ant` and/or `cheese` but not `fish`.
  - `and OR cheese` will only return results with one term or the other, not both. From our testing, this one only seemed to work effectively in Google.
  - `intitle:cheese` will only return results that have "cheese" in the main title of the page.

>   There are many other techniques you can use in various different search engines. Try seeing what others you can find!

Reviewer: Want to list these:

* https://support.google.com/websearch/answer/2466433?hl=en
* https://duckduckgo.com/duckduckgo-help-pages/results/syntax/
* https://support.microsoft.com/en-us/topic/advanced-search-options-b92e25f1-0085-4271-bdf9-14aaea720930
